### ISSUE TITLE
R 4.2.0: first argument has length > 1 error 

### ISSUE THREAD

**Issue by pat-s** [mlr-org/mlr3pipelines#661]
Ref: https://github.com/mlr-org/mlr3spatiotempcv/issues/170

``` r
library(mlr3)
library(mlr3learners)
library(mlr3pipelines)
library(mlr3spatiotempcv) # package is loaded but never used
library(paradox)

task <- tsk("mtcars")
learner_lm <- lrn("regr.lm")

g_ppl = ppl("targettrafo", graph = learner_lm)
g_ppl$param_set$values$targetmutate.trafo = function(x) log(x)
g_ppl$param_set$values$targetmutate.inverter = function(x) list(response = exp(x$response))

g_lrn <- GraphLearner$new(g_ppl)
g_lrn$train(task)
g_lrn$predict(task)
#> Error in get(mlr_reflections$task_types[type]$prediction): first argument has length > 1
#> This happened PipeOp targetinvert's $predict()
```

<sup>Created on 2022-04-28 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

## Context

R 4.2.0 is stricter WRT to sloppy subsetting. {mlr3spatiotempcv} adds an entry to `mlr_reflections$task_types` which let's the follwing subset calls in {mlr3pipelines} throw the error reported above.
There might be possibly more such instances in all of our pkgs which just turn into errors now (as of R 4.2.0).

https://github.com/mlr-org/mlr3pipelines/blob/e55b0a93b11c109534d5fd1db3b0d47a0a7e53ed/R/PipeOpTrafo.R#L385

https://github.com/mlr-org/mlr3pipelines/blob/e55b0a93b11c109534d5fd1db3b0d47a0a7e53ed/R/PipeOpRandomResponse.R#L124-L124

Possible solutions:

- Adding a `[1]` to the subset to make it explicit
- Remove the task_types in {mlr3spatiotempcv} (I don't think this should be the result of this issue)
- Find a more flexible solution than hardcoding `[1]`

**Comment by pat-s** [comment]
@s-kganz You have the following options:
- Use R < 4.2.0
- Use `remotes::install_github("mlr-org/mlr3pipelines@fix-661")`
- Don't load {mlr3spatiotempcv} 🙃

**Comment by s-kganz** [comment]
@pat-s RE option 3: awww, but it's so much fun! :)

I'll use the hotfix for now, I appreciate your quick response!

**Comment by mb706** [comment]
Closed by #662
